chalk-solve/src/recursive/fulfill.rs

```diff
@@ -323,7 +338,20 @@ impl<'s, I: Interner, Solver: SolveDatabase<I>, Infer: RecursiveInferenceTable<I
                 }
             },
             GoalData::EqGoal(EqGoal { a, b }) => {
-                self.unify(&environment, &a, &b)?;
+                self.unify(&environment, Variance::Invariant, &a, &b)?;
+            }
+            GoalData::SubtypeGoal(SubtypeGoal { a, b }) => {
+                match (a.ty(self.interner()), b.ty(self.interner())) {
+                    (Some(a_ty), Some(b_ty)) => {
+                        if a_ty.inference_var(self.interner()).is_some()
+                            && b_ty.inference_var(self.interner()).is_some()
+                        {
+                            return Err(NoSolution);
```

Comment:
This in particular is different than `chalk-engine`. There, we `Flounder`, here I didn't do all the refactoring to be able to handle this yet.

Reply:
I just set `self.cannot_solve =true` and continued processing. Probably not the correct thing to do. But I think that will also lead to `Floundering`.